### ISSUE TITLE
Feature Map doesn't show good performancs just as showed in the essay

### ISSUE THREAD

**Issue by HZIH** [uark-cviu/Micron-BERT#7]
I have plot a heatmap based on the features produced by the function forward_encoder which take two frames as input, however, the heatmap highlight most part of the img but not just the AU position.

**Comment by ZhengQinLai** [comment]
@HZIH, I'm in the same boat. I've also used the get_attn() function, which takes two frames as input. However, the results are not as promising as those presented in the paper. What's even more perplexing is when I input two identical apex frames into get_attn(), the produced heatmap is nearly identical to the ones I obtained when using apex frame and an onset frame as input.
![0199103aa81f3c56db2ed047344745c](https://github.com/uark-cviu/Micron-BERT/assets/87376138/33c61a9b-a7e0-4f55-8aab-a77f16263071)
![image](https://github.com/uark-cviu/Micron-BERT/assets/87376138/da49cf02-3a11-4d66-a259-51fce69ac6b0)

**Comment by ZhengQinLai** [comment]
This is my code. I have added some line to make it possible running correctly.
`python`

    def get_attmap(self, x, x_ref):
        #add
        x = self.backbone(x)
        x_ref = self.backbone(x_ref)

        x = x[:, 1:]
        x_ref = x_ref[:, 1:]
        #done
        features, attentions = self.att([x, x_ref, x])  # B h N N
        h = attentions.shape[2]
        nh = attentions.shape[1]
        att_h = att_w = int(h ** 0.5)
        diag_attentions = [attentions[:, :, i, i].unsqueeze(-1) for i in range(h)]
        diag_attentions = torch.cat(diag_attentions, axis=-1)
        diag_attentions = diag_attentions.reshape(-1, nh, att_h, att_w)

        patch_size = self.backbone.patch_embed.patch_size[0]
        diag_attentions = nn.functional.interpolate(
            diag_attentions,
            scale_factor=patch_size,
            mode="nearest",
        )

        diag_attentions *= 1 / (diag_attentions.max())
        #change
        #diag_attentions = 1 - diag_attentions
        #done
        return features, diag_attentions

**Comment by ZhengQinLai** [comment]
All right, they don't even use diag_attentions.

**Comment by HZIH** [comment]
> All right, they don't even use diag_attentions.

They have defined diag_attentions in their cross attention module, you can check it in the mae.py. The pretrained model shared to us is the version without diag_attentions, so you don't need to modify your code to transfer the original attention map to the diagonal.

**Comment by ZhengQinLai** [comment]
> > All right, they don't even use diag_attentions.
> 
> They have defined diag_attentions in their cross attention module, you can check it in the mae.py. The pretrained model shared to us is the version without diag_attentions, so you don't need to modify your code to transfer the original attention map to the diagonal.

So could it be that thw author uploading incorrect weights results in poor performance?

**Comment by HZIH** [comment]
> > > All right, they don't even use diag_attentions.
> > 
> > 
> > They have defined diag_attentions in their cross attention module, you can check it in the mae.py. The pretrained model shared to us is the version without diag_attentions, so you don't need to modify your code to transfer the original attention map to the diagonal.
> 
> So could it be that thw author uploading incorrect weights results in poor performance?

That's a possibility but I guess the data preprocessing and the code shared to us also cause the problem.

**Comment by ZhengQinLai** [comment]
I believe they don't employ any data preprocessing, as their attention map indicates that they don't even crop the faces. This is what piqued my curiosity about this work. I've written an email to the author and am eagerly awaiting his response.

**Comment by HZIH** [comment]
> I believe they don't employ any data preprocessing, as their attention map indicates that they don't even crop the faces. This is what piqued my curiosity about this work. I've written an email to the author and am eagerly awaiting his response.

If no data preprocessing method is applied, so many background noises exist, especially for the SMIC dataset. Except for the email, the author doesn't answer the issues in this repo, which makes me so confused.

**Comment by IcedDoggie** [comment]
> I believe they don't employ any data preprocessing, as their attention map indicates that they don't even crop the faces. This is what piqued my curiosity about this work. I've written an email to the author and am eagerly awaiting his response.

greetings. has any of the authors replied yet? 
Thank you

**Comment by ZhengQinLai** [comment]
> > I believe they don't employ any data preprocessing, as their attention map indicates that they don't even crop the faces. This is what piqued my curiosity about this work. I've written an email to the author and am eagerly awaiting his response.
> 
> greetings. has any of the authors replied yet? Thank you

No, none of the authors have replied yet. Also, I have given up on this paper and its methods.
